src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,320 @@
+
+    @Test
+    public void test_equals_sameUser_ok() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertEquals(user1, user2);
+    }
+
+    @Test
+    public void test_equals_differentUsers_notOk() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertNotEquals(user1, user2);
+    }
+
+    @Test
+    public void test_hashCode_equalUsers_equalHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_hashCode_differentUsers_differentHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(MIN_AGE);
+
+        assertNotEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_gettersAndSetters_workOk() {
+        User user = new User();
+        user.setLogin("user1");
+        user.setPassword("password");
+        user.setAge(MIN_AGE);
+        user.setId(1L);
+
+        assertEquals("user1", user.getLogin());
+        assertEquals("password", user.getPassword());
+        assertEquals(MIN_AGE, user.getAge());
+        assertEquals(FIRST_USER_ID, user.getId());
+
+        user.setLogin("newUser");
+        user.setPassword("newPassword");
+        user.setAge(MAX_AGE);
+        user.setId(2L);
+
+        assertEquals("newUser", user.getLogin());
+        assertEquals("newPassword", user.getPassword());
+        assertEquals(MAX_AGE, user.getAge());
+        assertEquals(SECOND_USER_ID, user.getId());
+    }
+
+    /* Tests in class RegistrationServiceImpl */
+
+    @Test
+    public void register_existingUser_throwsException() {
+        User user1 = new User();
+        user1.setLogin("Recruit");
+        user1.setPassword("password");
+        user1.setAge(MIN_AGE);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("Recruit");
+        user2.setPassword("KingBoss");
+        user2.setAge(MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user2));
```

Comment:
checklist issue, you can't rely on the method you test in the test of this method 🤯
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/2f6edf45-e804-4d07-80ab-75dadf054a6f)